File: pset1/mario/less/mario.c
#include <stdbool.h>
#include <stdio.h>
#include <cs50.h>

// generate pyramid based on input height

void make_pyramid(int height);
int prompt_height(int max);

// main
int main(void)
{

    // prompt user for pyramid height
    int height = prompt_height(8);
    // print pyramid
    make_pyramid(height);
}

// print pyramid
void make_pyramid(int height)
{
    int q = height - 1;
    for (int j = 0; j < height; j++)
    {
        for (int i = 0; i < height; i++)
        {
            if (i >= q)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        q = q - 1;
        printf("\n");
    }

}

// prompt user for height
int prompt_height(int max)
{

    int height = get_int("Height: ");
    while (height > max || height < 1)
    {
        height = get_int("Height: ");
    }
    return height;
}

File: pset6/mario/less/mario.py
from cs50 import get_int

# main function


def main():
    height = prompt_height(8)
    make_pyramid(height)


# function to promt height from user
def prompt_height(maxheight: int):
    height = 0

    while (height > maxheight or height < 1):
        height = get_int("Height:")
    return height


# function to build and print pyramid
def make_pyramid(height: int):
    # nr of spaces required in a row
    q = height - 1
    # iterate through rows
    for j in range(height):
        # iterate through columns
        for i in range(height):
            # if location column location >= nr of spaces in that row print #
            if i >= q:
                print("#", end='')
            # if location column location <= nr of spaces in that row print space
            else:
                print(" ", end='')
            i += 1
        # print new line
        print("")
        # adjust number of spaces required in new row
        q -= 1
        j += 1


# run main function
if __name__ == '__main__':
    main()
